### ISSUE TITLE
Pedestrians on crosswalks

### ISSUE THREAD

**Issue by burnout2k** [lgsvl/simulator#1859]
Hello

Pedestrians seem not to follow directions on crosswalks when it comes to random traffic mode at times.

Here is how the crosswalk is set up:
![crosswalks](https://user-images.githubusercontent.com/35662015/140628740-b8d6ad8a-61e6-4d93-97b2-864f9d3e804d.png)

And here is how it plays out:
https://youtu.be/74OdB1-b50k

Is it normal for Unity agents to go "crazy" or am I doing something wrong?
At some crossroads, they walk away like 30 meters (100 feet) before they take the turn to cross the road.
Though at the same time on other crosswalks they act exactly as assigned.
They do follow the traffic light status (eg not stepping on the road with a red light).

Everything was tested with the latest version of the simulation.

**Comment by EricBoiseLGSVL** [comment]
Yes, we have a few bugs that we are looking into.  Thanks for the post.  We hope to have this fixed in the next release. The skating as well :)
